astro_3.0/features/BountiesV2/BountiesV2.tsx

```diff
@@ -0,0 +1,238 @@
+
+import { useUpdateEffect } from 'react-use';
+
+import axios from 'axios';
+import { Icon } from 'components/Icon';
+import { appConfig } from 'config';
+import { NoResultsView } from 'astro_2.0/components/NoResultsView';
+import { Loader } from 'components/loader';
+import {
+  BountyIndex,
+  OpenSearchQuery,
+  OpenSearchResponse,
+} from 'services/SearchService/types';
+import { BountiesList } from './BountiesList';
+
+import styles from './BountiesV2.module.scss';
+
+const PAGING_SIZE = 50;
+
+export type BountiesData = { total: number; bounties: BountyIndex[] };
+
+export const fetchBounties = async (
+  filter: Filter,
+  from = 0
+): Promise<BountiesData> => {
+  const baseUrl = process.browser
+    ? window.APP_CONFIG.SEARCH_API_URL
+    : appConfig.SEARCH_API_URL;
+
+  const tagsQueries =
+    filter.tags?.map(tag => ({
+      match: {
+        tags: tag,
+      },
+    })) ?? ([] as Record<string, unknown>[]);
+
+  const statusQueries =
+    Object.entries(filter.statuses)
+      .filter(([, value]) => value)
+      .map(([key]) => ({
+        match: {
+          proposalStatus: key,
+        },
+      })) ?? ([] as Record<string, unknown>[]);
+
+  const query: OpenSearchQuery = {
+    bool: {
+      must: [
+        {
+          bool: {
+            should: tagsQueries,
+          },
+        },
+        {
+          bool: {
+            should: statusQueries,
+          },
+        },
+      ],
+    },
+  };
+
+  if (filter.daoId) {
+    query.bool?.must?.push({
+      match: {
+        daoId: filter.daoId,
+      },
+    });
+  }
+
+  const res = await axios.post<unknown, { data: OpenSearchResponse }>(
+    `${baseUrl}/bounty/_search`,
+    {
+      from,
+      size: PAGING_SIZE,
+      query,
+      sort: {
+        creatingTimeStamp: {
+          order: 'desc',
+        },
+      },
+    }
+  );
+
+  const bountyIndexes =
+    res.data?.hits?.hits?.map(hit => {
+      const { _source: rawBounty } = hit;
+
+      return rawBounty;
+    }) ?? [];
+
+  return {
+    total: res.data?.hits?.total.value,
+    bounties: bountyIndexes as BountyIndex[],
+  };
+};
+
+type Filter = {
+  daoId: string;
+  tags: string[];
+  statuses: {
+    InProgress: boolean;
+    Approved: boolean;
+    Rejected: boolean;
+  };
+};
+
+const STATUS_LIST = {
+  'In Progress': 'InProgress',
+  Approved: 'Approved',
+  Rejected: 'Rejected',
+};
+
+interface Props {
+  initialData: BountiesData | null;
+}
+
+export const BountiesV2: VFC<Props> = ({ initialData }) => {
+  const [bounties, setBounties] = useState<BountyIndex[]>(
+    initialData?.bounties ?? []
+  );
+
+  const [error, setError] = useState();
+  const [paginationTotal, setPaginationTotal] = useState(
+    initialData?.total ?? 0
+  );
+  const [filter, setFilter] = useState<Filter>({
+    daoId: '',
+    tags: [],
+    statuses: {
+      InProgress: false,
+      Approved: false,
+      Rejected: false,
+    },
+  });
+
+  useUpdateEffect(() => {
+    fetchBounties(filter)
```

Comment:
Why cant we use SWR hook here to fetch data? We will get caching and auto revelidate depending on filters.

Reply:
Good catch, let me look into that.

Reply:
@VERZUOL1 Updated to use `useSWRInfinite`, please review again 🙏